core/src/Kokkos_Complex.hpp

```diff
@@ -631,8 +631,7 @@ RealType real (const complex<RealType>& x) {
 template<class RealType>
 KOKKOS_INLINE_FUNCTION
 RealType abs (const complex<RealType>& x) {
-  // FIXME (mfh 31 Oct 2014) Scale to avoid unwarranted overflow.
-  return std::sqrt (real (x) * real (x) + imag (x) * imag (x));
+  return std::hypot(x.real(),x.imag());
```

Comment:
Hm, sorry, I forgot that this won't work with CUDA.  However, CUDA provides `hypot` in the global namespace, so I think we can do what `Kokkos::ArithTraits` does for e.g., `isinf`.
```c++
template<class RealType>
KOKKOS_INLINE_FUNCTION RealType
abs (const complex<RealType>& z) {
#ifndef __CUDA_ARCH__
  using std::hypot;
#endif
  return hypot (z.real (), z.imag ());
}
```